### PR TITLE
Refactor `activate` to avoid complexity in the main method

### PR DESCRIPTION
**Pull request by 00vareladavid**
More idiomatic too, I think

### CODE REVIEW

File: src/API.jl [hunk 1]
```diff
@@ -569,29 +569,28 @@ function status(ctx::Context, mode=PKGMODE_PROJECT)
...
-        # 3. activate the non-existing directory (e.g. as in `pkg> activate .` for initing a new env)
-        if Types.isdir_windows_workaround(path)
-            path = abspath(path)
-        elseif devpath !== nothing
-            path = abspath(devpath)
-        else
-            path = abspath(path)
-        end
+activate() = activate(nothing)
+activate(path::Nothing) = (Base.ACTIVE_PROJECT[] = Base.load_path_expand(path))
```
Comment (fredrikekre) [inline src/API.jl]: I don't think this method needs to be exposed, perhaps just replace these two methods with
```
activate() = (Base.ACTIVE_PROJECT[] = Base.load_path_expand(nothing))
```
and update https://github.com/JuliaLang/Pkg.jl/blob/06c96968e2216fecbf873460db135aea711d0e28/src/REPLMode.jl#L606 to 
```
return API.activate() 
```
?

Comment (00vareladavid) [reply src/API.jl]: Ok, that seems nicer to me too

File: src/API.jl [hunk 2]
```diff
@@ -569,29 +569,27 @@ function status(ctx::Context, mode=PKGMODE_PROJECT)
...
-        # 2. if path exists in deps, and the dep is deved, activate that path (`devpath` above)
-        # 3. activate the non-existing directory (e.g. as in `pkg> activate .` for initing a new env)
-        if Types.isdir_windows_workaround(path)
-            path = abspath(path)
-        elseif devpath !== nothing
-            path = abspath(devpath)
-        else
-            path = abspath(path)
-        end
+activate() = (Base.ACTIVE_PROJECT[] = Base.load_path_expand(nothing))
```
Comment (Evizero) [inline src/API.jl]: isn't  `Base.load_path_expand(nothing))` simply `nothing` ?

Comment (00vareladavid) [reply src/API.jl]: Yes, I hadn't considered that. But given that `activate` is already coupled with `load_path_expand`, I think it makes sense to be explicit and avoid "magic values".

**Pull request merged**